core/src/main/java/org/apache/calcite/sql/dialect/SnowflakeSqlDialect.java

```diff
@@ -588,6 +605,53 @@ private SqlNode getIdentifier(SqlBasicCall intervalOperand) {
     return ((SqlBasicCall) aggCall).operand(0);
   }
 
+  private void formatDate(SqlWriter writer, SqlCall call, int leftPrec, int rightPrec) {
+    if (call.operand(0).toString().equals("'EEEE'") || call.operand(0).toString().equals("'E4'")) {
+      SqlCall toVarcharNode = createVarCharNode(call);
+      List<String> abvList = Arrays.asList("Sun", "Mon", "Tue", "Wed", "Thu", "Fri", "Sat");
+      ArrayList<String> abvDays = new ArrayList<>(abvList);
```

Comment:
Can we iterate on List<String> abvList directly. Do we need another list here ?

Reply:
we can write it like this - new ArrayList<>(Arrays.asList("Sun", "Mon", "Tue", "Wed", "Thu", "Fri", "Sat"))

Reply:
done